src/main.ts

```diff
@@ -130,18 +130,25 @@ async function run(): Promise<void> {
     }
 
     if (config.vulnerability_check) {
+      core.setOutput('vulnerable-changes', JSON.stringify(vulnerableChanges))
```

Comment:
Is `stringify` what we want to be using if we expect people to consume the JSON?
```
vulnName = "Security Advisory for \"Log4Shell\""
=> 'Security Advisory for "Log4Shell"'
JSON.stringify(vulnName)
=> '"Security Advisory for \\"Log4Shell\\""' 
```

When I try to use it in `sh`:
```
~/w
$ cat > foo.json
[{ "name": '"Security Advisory for \\"Log4Shell\\""' }]
^D
~/w
$ cat foo.json | jq .
parse error: Invalid literal at line 1, column 13
```

If we plan to go down this route, let's move the "stringification" of output to an external function that includes unit tests.

Reply:
`vulnerableChanges` as well as others is a JS object parsed from a JSON API response and `stringify` here just serializes it to a string.

In your example `JSON.stringify(vulnName) => '"Security Advisory for \\"Log4Shell\\""'` shows extra quotes because it's a JS string value (`'`) with a JSON string inside (`"`) with some escaped quotes inside

In the shell test you store an invalid JSON by keeping those single quotes. Let's do this with the whole object that you use in the second part:

```
$ node
> obj = [{ name: "Security Advisory for \"Log4Shell\"" }]
[ { name: 'Security Advisory for "Log4Shell"' } ]
> JSON.stringify(obj)
'[{"name":"Security Advisory for \\"Log4Shell\\""}]'
> fs.writeFileSync('foo.json', JSON.stringify(obj))
^D

$ cat foo.json
[{"name":"Security Advisory for \"Log4Shell\""}]

$ cat foo.json | jq
[
  {
    "name": "Security Advisory for \"Log4Shell\""
  }
]

$ cat foo.json | jq '.[].name'
"Security Advisory for \"Log4Shell\""

$ cat foo.json | jq '.[].name' --raw-output
Security Advisory for "Log4Shell"
```

As you see, it produces JSON data that can be parsed again.

Reply:
@laughedelic thanks for the explanation 🙇 

Do you see a way in which we could escape these results so people reading directly from the object's property don't get bitten by cases like `{"foo": "a'b"}` as you pointed out above?

Reply:
☝️ came to the same conclusion in a 🧵 above. feels too delicate in the current form to me, but I'm not super familiar with TS so I'll keep an open mind here 😂